pyscripts/build_docker_image.py

```diff
@@ -10,193 +10,237 @@
+        )
+        args = parser.parse_args()
+
+        if args.repo == default_repo:
+            if args.language == "pythonpreview":
+                args.repo = "Azure/azure-iot-sdk-python-preview"
+            else:
+                args.repo = "Azure/azure-iot-sdk-" + args.language
+            print(Fore.YELLOW + "Repo not specified.  Defaulting to " + args.repo)
+
+        print_separator = "".join("/\\" for _ in range(80))
+
+        auth_config = {
+            "username": os.environ["IOTHUB_E2E_REPO_USER"],
+            "password": os.environ["IOTHUB_E2E_REPO_PASSWORD"],
+        }
+
+
+        def get_dockerfile_directory(tags):
+            script_dir = os.path.dirname(os.path.realpath(__file__))
+            return os.path.normpath(os.path.join(script_dir, "../ci-wrappers/" + tags.language))
+
+
+        def print_filtered_docker_line(lineobj):
+            lineobj = lineobj.splitlines()
+            for line in lineobj:
+                try:
+                    linecode = line.strip().decode("utf-8")
+                    linecode = linecode[:linecode.rfind("\r\n")] + '}'
+                    jline = json.loads(linecode)
+                    obj = jline["stream"]
+                except:
+                    print(''.join([chr(i) if i < 128 else '\x23' for i in line]))
                 else:
-                    pass
+                    if "status" in obj:
+                        if "id" in obj:
+                            if obj["status"] not in [
+                                "Waiting",
+                                "Downloading",
+                                "Verifying Checksum",
+                                "Extracting",
+                                "Preparing",
+                                "Pushing",
+                            ]:
+                                print("{}: {}".format(obj["status"], obj["id"]))
+                            else:
+                                pass
+                        else:
+                            print(obj["status"])
+                    elif "error" in obj:
+                        print ("docker error: {}".format(line))
+                        raise Exception(obj["error"])
+                    elif "Step" in obj or "---" in obj:
+                        print("{}".format(obj).strip())
+                    elif obj == "\n":
+                        pass
+                    else:
+                        lineout = "".format("{}", line)
+                        print(''.join([chr(i) if i < 128 else '\x23' for i in lineout]))
+        
+
+        def build_image(tags):
+            print(print_separator)
+            print("BUILDING IMAGE")
+            print(print_separator)
+
+            force_flag = 0
+
+            if sys.platform == 'win32':
+                base_url = "tcp://127.0.0.1:2375"
+            else:
+                base_url = "unix://var/run/docker.sock"
+            api_client = docker.APIClient(base_url=base_url)
```

Comment:
these 5 lines are repeated over and over (setting base_url and creating the api_client).  Can you make these global so there's not so much copy/paste.  (at least set base_url once, but preferably create api_client only once. )

Reply:
Fixed. made base_url global